nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -1,21 +1,146 @@
-"""Authors: Cody Baker."""
+"""Authors: Saksham Sharda, Cody Baker."""
 from pathlib import Path
+from typing import Union, Tuple
+
 import numpy as np
-from typing import Union
+from tqdm import tqdm
+
+from ....utils.json_schema import FilePathType
 
 try:
     import cv2
 
     HAVE_OPENCV = True
 except ImportError:
     HAVE_OPENCV = False
+INSTALL_MESSAGE = "Please install opencv to use this interface! (pip install opencv-python)"
 
 PathType = Union[str, Path]
 
 
+class VideoCaptureContext:
+    """Retrieving video metadata and frames using a context manager"""
+
+    def __init__(self, file_path: FilePathType):
+        assert HAVE_OPENCV, INSTALL_MESSAGE
+        self.vc = cv2.VideoCapture(filename=file_path)
+        self.file_path = file_path
+        self._current_frame = 0
+        self._frame_count = None
+        self._movie_open_msg = "The Movie file is not open!"
+
+    def get_movie_timestamps(self):
+        """Return numpy array of the timestamps(s) for a movie file."""
+        ts2 = []
+        for no in tqdm(range(self.get_movie_frame_count()), desc="retrieving timestamps"):
+            success, frame = self.vc.read()
+            ts2.append(self.vc.get(cv2.CAP_PROP_POS_MSEC) / 1000)
+            if not success:
+                break
+        return np.array(ts2)
+
+    def get_movie_fps(self):
+        """Return the internal frames per second (fps) for a movie file"""
+        assert self.isOpened(), self._movie_open_msg
+        prop = self.get_cv_attribute("CAP_PROP_FPS")
+        return self.vc.get(prop)
+
+    def get_frame_shape(self) -> Tuple:
+        """Return the shape of frames from a movie file."""
+        frame = self.get_movie_frame(0)
+        if frame is not None:
+            return frame.shape
+
+    @property
+    def frame_count(self):
+        if self._frame_count is None:
+            self._frame_count = self._movie_frame_count()
+        return self._frame_count
+
+    @frame_count.setter
+    def frame_count(self, val):
+        self._frame_count = val
```

Comment:
```suggestion
    @frame_count.setter
    def frame_count(self, val: int):
        assert val > 0, "You must set a positive frame_count (received {val})."
        assert val <= self._movie_frame_count(), "Cannot set manual frame_count beyond length of video ({val})."
        self._frame_count = val
```
One main use of `@property.setter` functions is to provide some type of input validation for manually overriding things that are otherwise internal/private values.